src/actions/LoginActions.tsx

```diff
@@ -83,19 +83,20 @@ export function initializeAccount(navigation: NavigationBase, account: EdgeAccou
 
         // New user FIO handle registration flow (if env is properly configured)
         const { freeRegApiToken = '', freeRegRefCode = '' } = typeof ENV.FIO_INIT === 'object' ? ENV.FIO_INIT : {}
+        const createWalletsPromise = createCustomWallets(account, fiatCurrencyCode, selectedEdgetokenIds, dispatch)
+
         if (freeRegApiToken !== '' && freeRegRefCode !== '') {
-          await Promise.all([
-            createCustomWallets(account, fiatCurrencyCode, selectedEdgetokenIds, dispatch),
-            Airship.show<boolean>(bridge => <FioCreateHandleModal bridge={bridge} />).then(isCreateHandle => {
-              if (isCreateHandle) navigation.navigate('fioCreateHandle', { freeRegApiToken, freeRegRefCode })
+          Airship.show<boolean>(bridge => <FioCreateHandleModal bridge={bridge} createWalletsPromise={createWalletsPromise} />)
+            .then(isCreateHandle => {
+              if (isCreateHandle) {
+                navigation.navigate('fioCreateHandle', { freeRegApiToken, freeRegRefCode })
+              }
             })
-          ]).then(async ([_customWallets, _fioCreate]) => {
-            return await updateWalletsRequest()(dispatch, getState)
-          })
-        } else {
-          await createCustomWallets(account, fiatCurrencyCode, selectedEdgetokenIds, dispatch)
-          await updateWalletsRequest()(dispatch, getState)
+            .catch(showError)
         }
+
+        await createWalletsPromise
```

Comment:
Yikes. you'll be calling updateWalletsRequest before the fio wallets are created. I would await the Airship.show() instead and do away with .then()